### PR TITLE
[BugFix] Fix Struct serialization bug

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

In this pr:
1. We don't serialize Struct's field name anymore, because the field name is created from TypeDescriptor, and doesn't need to transfer by the network.
2. Fix struct serialization error implementation.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/column/struct_column.cpp [hunk 1]
```diff
@@ -223,55 +219,61 @@ void StructColumn::append_default(size_t count) {
...
+uint32_t StructColumn::max_one_element_serialize_size() const {
+    // TODO: performance optimization.
+    size_t n = size();
+    uint32_t max_size = 0;
+    for (size_t i = 0; i < n; i++) {
+        uint32_t row_max_size = 0;
+        for (const auto& column : _fields) {
+            row_max_size += column->max_one_element_serialize_size();
+        }
+        max_size = std::max(max_size, row_max_size);
```
Comment (zombee0) [inline be/src/column/struct_column.cpp:267]: I think in this code the row_max_size is the sum size of max_element in each fields, so the the outer loop will get the same row_max_size.

Comment (Smith-Cruise) [reply be/src/column/struct_column.cpp]: fixed

**Review by zombee0 — APPROVED** [review]
LGTM

**Review by dirtysalt — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**